src/PointwiseFunctions/AnalyticSolutions/Elasticity/HalfSpaceMirror.cpp

```diff
@@ -2,234 +2,180 @@
+
+    const double surface_term_zz =
+        shear_modulus / (lame_parameter + shear_modulus);
+    const double strain_zz =
+        -0.25 / (shear_modulus * M_PI) *
+        integration(
+            [&r, &z, &surface_term_zz, this](const double k) noexcept {
+              return k * gsl_sf_bessel_J0(k * r) *
+                     exp(-k * z - square(k * beam_width_ / 2.)) *
+                     (surface_term_zz + k * z);
+            },
+            lower_boundary, absolute_tolerance_, relative_tolerance_);
+
+    if (r >= 100 * std::numeric_limits<double>::epsilon()) {
+      const double strain_rz =
+          -0.25 / (shear_modulus * M_PI) *
+          integration(
+              [&r, &z, this](const double k) noexcept {
+                return k * gsl_sf_bessel_J1(k * r) *
+                       exp(-k * z - square(k * beam_width_ / 2.)) * (k * z);
+              },
+              lower_boundary, absolute_tolerance_, relative_tolerance_);
+
+      const double surface_term_r = 1. - (lame_parameter + 2. * shear_modulus) /
+                                             (lame_parameter + shear_modulus);
+      const double displacement_r =
+          0.25 / (shear_modulus * M_PI) *
+          integration(
+              [&r, &z, &surface_term_r, this](const double k) noexcept {
+                return gsl_sf_bessel_J1(k * r) *
+                       exp(-k * z - square(k * beam_width_ / 2.)) *
+                       (surface_term_r + k * z);
+              },
+              lower_boundary, absolute_tolerance_, relative_tolerance_);
+      const double cos_phi = get<0>(x)[i] / r;
+      const double sin_phi = get<1>(x)[i] / r;
+      const double strain_pp = displacement_r / r;
+      const double strain_rr = trace_term - strain_pp - strain_zz;
+      get<0, 0>(strain)[i] =
+          strain_pp + square(cos_phi) * (strain_rr - strain_pp);
+      get<0, 1>(strain)[i] = cos_phi * sin_phi * (strain_rr - strain_pp);
+      get<0, 2>(strain)[i] = cos_phi * strain_rz;
+      get<1, 1>(strain)[i] =
+          strain_pp + square(sin_phi) * (strain_rr - strain_pp);
+      get<1, 2>(strain)[i] = sin_phi * strain_rz;
+      get<2, 2>(strain)[i] = strain_zz;
+
     } else {
-      auto displacement_w = integration(
-          [&r, &z, &shear_modulus, &lame_parameter, this](const double k) {
-            return applied_force_ / (2. * shear_modulus) *
-                   gsl_sf_bessel_J1(k * r) * exp(-k * z) *
-                   (1 -
-                    (lame_parameter + 2. * shear_modulus) /
-                        (lame_parameter + shear_modulus) +
-                    k * z) *
-                   1 / (2. * M_PI) * exp(-square(k * beam_width_ / 2.));
-          },
-          lower_boundary, absolute_tolerance_);
-
-      cos_phi = get<0>(x)[i] / w[i];
-      sin_phi = get<1>(x)[i] / w[i];
-      strain_pp = displacement_w / w[i];
-      strain_rr = theta - strain_pp - strain_zz;
+      get<0, 0>(strain)[i] = 0.5 * (trace_term - strain_zz);
```

Comment:
Please add a comment here and in the displacement function that for r=0 the remaining components are left at 0.